### PR TITLE
Export fastify.SUPPORTED_METHODS

### PR DESCRIPTION
**Pull request by allevo**
As titled.

In some circumstance, the developer needs to know which methods fastify supports.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

Add typescript definition too

### CODE REVIEW

**Comment by corbinu** [comment]
I have to say I’m -1 on this. TypeScript and Flowlang fill this role

**Comment by allevo** [comment]
> Isn't already stated [here](https://github.com/fastify/fastify/blob/master/docs/Routes.md#full-declaration9?

@delvedor We haven't the list. Only the documentation lists the methods.

> I have to say I’m -1 on this. TypeScript and Flowlang fill this role

@corbinu I'm bit lost. I like to have the array in order to use it at runtime. The type check is not the aim of this PR

File: fastify.d.ts [hunk 1]
```diff
@@ -385,6 +385,11 @@ declare namespace fastify {
      * Prints the representation of the internal radix tree used by the router
      */
     printRoutes(): string
+
+    /**
+     * Array that contains all HTTP methods
+     */
+    SUPPORTED_METHODS: [string]
```
Comment (evanshortiss) [inline fastify.d.ts]: This should be `string[]` or `Array<string>`. Current syntax would denote an array with just a single string entry

Comment (allevo) [reply fastify.d.ts]: Fixed

**Comment by corbinu** [comment]
@allevo why not just calculate this at runtime then?

**Comment by allevo** [comment]
> @allevo why not just calculate this at runtime then?

How?

**Comment by corbinu** [comment]
http.METHODS or am I missing something?

**Comment by allevo** [comment]
> http.METHODS or am I missing something?

Not all methods are supported. For example `TRACE`

**Comment by corbinu** [comment]
ya I guess I would just hard code this but totally get if others want it.

**Comment by allevo** [comment]
NB: `find-my-way` has another array 

https://github.com/delvedor/find-my-way/blob/cc6ac4e5a1b1617ca2d84cce33201f727ae12c8b/index.js#L16

**Comment by mcollina** [comment]
> Not all methods are supported. For example TRACE

I would add support for all methods in `require('http')`.

**Comment by allevo** [comment]
@mcollina after https://github.com/delvedor/find-my-way/pull/72/files

**Comment by mcollina** [comment]
after #855

**Comment by mcollina** [comment]
On a second thoughts.. why are we exposing this on the instance and not on the module itself?

**Comment by allevo** [comment]
I'm closing this PR. All known HTTP methods are supported. This is not needed anymore

**Comment by delvedor** [comment]
> I'm closing this PR. All known HTTP methods are supported. This is not needed anymore

That's not true.
These are the methods we are currently supporting:
https://github.com/fastify/fastify/blob/5fefda9dd2f882e807b7ae736d39be4860c13d6b/fastify.js#L13

`find-my-way` supports all http methods, but Fastify is not.

**Pull request closed without merging**